src/io/flutter/inspector/InspectorService.java

```diff
@@ -240,43 +232,29 @@ private void onVmServiceReceived(String streamId, Event event) {
    * new frames will be triggered to draw unless something changes in the UI.
    */
   public CompletableFuture<Boolean> isWidgetTreeReady() {
-    if (useDaemonApi()) {
```

Comment:
the invokeServiceMethodObservatory code path is unfortunately using 
eval instead of invoking the vm service extensions via the right api. That code path should only be used when paused at a breakpoint when we cannot invoke the service extensions or for cases where we really need to be using eval due to needing to get back actual VM Service api instance refs for objects.
What I think we want instead of this CL is to invoke the VM service extensions via the VMService's api to invoke vm service extensions on an isolate instead of invoking them on a daemon protocol. :(

Reply:
OK, I didn't follow that completely, but I gather we still want to avoid using the daemon protocol for the inspector traffic (it'll fix some existing issues, and likely be faster).

I think I'll close this PR; we still have the existing issue to avoid use of the daemon protocol. I'll re-create a smaller version of this PR with the workaround for the package:dwds issue (it'll still be correct even after the dwds fix lands).

Reply:
yes. we want to avoid the daemon protocol but continue to call the ext.flutter.inspector service extensions. Just use the VMService to call them directly instead of the daemon.
the invokeServiceMethodObservatory code path is weird mainly legacy code using eval to invoke the InspectorService.